spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/File.scala

```diff
@@ -220,15 +235,24 @@ private[bigdl] class FileReader(fileName: String) {
   private val conf = File.getConfiguration(fileName)
   private val path = new Path(fileName)
   private val fs: FileSystem = path.getFileSystem(conf)
+  private[bigdl] val httpPrefix: String = "http"
 
   /**
    * get an InputStream
    * @return
    */
   def open(): InputStream = {
     require(inputStream == null, s"File $fileName has been opened already.")
-    require(fs.exists(path), s"$fileName is empty!")
-    inputStream = fs.open(path)
+
+    if(!fileName.startsWith(httpPrefix)) {
+      require(fs.exists(path), s"$fileName is empty!")
+      inputStream = fs.open(path)
+    } else {
+      File.downloadFromUrl(fileName, path.toString.replace(path.getName, ""))
```

Comment:
Are you sure this work? Looks like path is create from fileName.

private val path = new Path(fileName)